examples/examples/tower_http.rs

```diff
@@ -0,0 +1,113 @@
+use jsonrpsee::http_client::HttpClientBuilder;
+use jsonrpsee::http_server::{HttpServerBuilder, HttpServerHandle, RpcModule};
+
+/// Define a custom logging mechanism to detect the time passed
+/// between receiving the request and proving the response.
+///
+/// The implementation relies upon [logger::HttpLogger].
+#[derive(Clone)]
+struct Timings;
+
+impl logger::HttpLogger for Timings {
+	type Instant = Instant;
+
+	fn on_request(&self, remote_addr: SocketAddr, request: &Request<Body>) -> Self::Instant {
+		println!("[Logger::on_request] remote_addr {}, request: {:?}", remote_addr, request);
+		Instant::now()
+	}
+
+	fn on_call(&self, name: &str, params: Params, kind: logger::MethodKind) {
+		println!("[Logger::on_call] method: '{}', params: {:?}, kind: {}", name, params, kind);
+	}
+
+	fn on_result(&self, name: &str, success: bool, started_at: Self::Instant) {
+		println!("[Logger::on_result] '{}', worked? {}, time elapsed {:?}", name, success, started_at.elapsed());
+	}
+
+	fn on_response(&self, result: &str, started_at: Self::Instant) {
+		println!("[Logger::on_response] result: {}, time elapsed {:?}", result, started_at.elapsed());
+	}
+}
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+	tracing_subscriber::FmtSubscriber::builder()
+		.with_env_filter(tracing_subscriber::EnvFilter::from_default_env())
+		.try_init()
+		.expect("setting default subscriber failed");
+
+	let (addr, _handler) = run_server().await?;
+	let url = format!("http://{}", addr);
+
+	let client = HttpClientBuilder::default().build(&url)?;
+	let response: String = client.request("say_hello", None).await?;
+	println!("[main]: response: {:?}", response);
+	let _response: Result<String, _> = client.request("unknown_method", None).await;
+	let _ = client.request::<String>("say_hello", None).await?;
+
+	Ok(())
+}
+
+async fn run_server() -> anyhow::Result<(SocketAddr, HttpServerHandle)> {
+	// Custom tower service to handle the RPC requests
+	let builder = tower::ServiceBuilder::new()
+		// Add high level tracing/logging to all requests
+		.layer(
+			TraceLayer::new_for_http()
+				.on_body_chunk(|chunk: &Bytes, latency: Duration, _: &tracing::Span| {
+					tracing::trace!(size_bytes = chunk.len(), latency = ?latency, "sending body chunk")
+				})
+				.make_span_with(DefaultMakeSpan::new().include_headers(true))
+				.on_response(DefaultOnResponse::new().include_headers(true).latency_unit(LatencyUnit::Micros)),
+		)
+		// Mark the `Authorization` request header as sensitive so it doesn't show in logs
+		.layer(SetSensitiveRequestHeadersLayer::new(once(hyper::header::AUTHORIZATION)))
+		.timeout(Duration::from_secs(2));
+
+	let server = HttpServerBuilder::new().set_logger(Timings).build("127.0.0.1:0".parse::<SocketAddr>()?).await?;
+	let addr = server.local_addr()?;
+
+	let mut module = RpcModule::new(());
+	module.register_method("say_hello", |_, _| Ok("lo")).unwrap();
+
+	let handler = server.start_with_builder(module, builder)?;
```

Comment:
I suggest to rename this method to `Server::start_with_middleware` or something like that because I kind of want not confuse it with the `HttpServerBuilder`

Besides that it looks great now, perhaps remove the `HttpLogger` stuff in the example to keep it easier to follow that it's not really related to the middleware or rename the example then document that the `tower middleware` and `HttpLogger` are not mutually exclusive.